Source/WebCore/loader/DocumentLoader.cpp

```diff
@@ -2140,7 +2140,14 @@ void DocumentLoader::startLoadingMainResource()
     if (contentFilterInDocumentLoader())
         m_contentFilter = !m_substituteData.isValid() ? ContentFilter::create(*this) : nullptr;
 #endif
-    
+
+    auto url = m_request.url();
+    auto fragmentDirective = url.consumeFragmentDirective();
+    m_request.setURL(url, m_request.didFilterLinkDecoration());
+    auto page = m_frame->protectedPage();
+    if (m_frame && page && !fragmentDirective.isEmpty() && !fragmentDirective.isNull())
```

Comment:
`m_frame` is guaranteed to be non-null here, assuming it is non-null in `auto page = m_frame->protectedPage();`